src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+matrix.
+"""
+function Base.:*(σ::Real, ρ::UnivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(zero(σ), σ, ρ)
+end
+
+function Base.:*(σ::Real, ρ::Distribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(size(ρ)), σ, ρ)
+end
+
+function Base.:*(σ::AbstractMatrix{<:Real}, ρ::MultivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    elseif size(σ)[end] ≠ size(ρ)[1]
+        throw(DimensionMismatch(
+            "scaling matrix and distribution have incompatible dimensions" *
+            "$(size(σ)) and $(size(ρ)))"
+        ))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(σ), σ, ρ)
+end
+Base.:*(σ::AbstractVector{<:Real}, ρ::MultivariateDistribution) = Diagonal(σ) * ρ
+
+
+# Composing affine distributions
+
+function Base.:+(μ::Real, d::UnivariateAffine)
+    return AffineDistribution(μ + d.μ, d.σ, d.ρ)
+end
+function Base.:+(μ::AbstractArray, d::MultivariateAffine)
+    return AffineDistribution(μ + d.μ, d.σ, d.ρ)
+end
+
+
+function Base.:*(σ::Real, d::UnivariateAffine)
+    return AffineDistribution(σ * d.μ, σ * d.σ, d.ρ)
+end
+
+function Base.:*(σ::Real, d::MultivariateAffine)
+    return AffineDistribution(σ * d.μ, σ * d.σ, d.ρ)
+end
+
+function Base.:*(σ::AbstractMatrix, d::MultivariateAffine)
+    return AffineDistribution(σ * d.μ, σ * d.σ, d.ρ)
+end
+
+
+Base.:+(d::NonMatrixDistribution, μ::ArrayLike) = μ + d
+Base.:-(d::NonMatrixDistribution) = -1 * d
+Base.:-(d::NonMatrixDistribution, μ::ArrayLike) = -μ + d
+Base.:-(μ::ArrayLike, d::NonMatrixDistribution) = μ + -d
+
+Base.:*(d::NonMatrixDistribution, σ::Real) = σ * d
+Base.:/(d::NonMatrixDistribution, τ::Real) = Base.inv(τ) * d
+Base.:\(τ::ArrayLike, d::NonMatrixDistribution) = Base.inv(τ) * d
+
+
+#### Extremes
+
+## Univariate
+
+function maximum(d::UnivariateAffine)
+    maxim = d.σ > 0 ? maximum(d.ρ) : minimum(d.ρ)
+    return d.μ + d.σ * maxim
+end
+
+function minimum(d::UnivariateAffine)
+    minim = d.σ > 0 ? minimum(d.ρ) : maximum(d.ρ)
+    return d.μ + d.σ * minim
+end
+
+function extrema(d::UnivariateAffine)
+    extremes = extrema(d.ρ)
+    extremes = _flip(d.σ, extremes)
+    return @. d.μ + d.σ * extremes
+end
+
+
+## Multivariate
+# when finding the extrema, we need to flip the extremes if the scale is negative
+_flip(signed::Real, tuple::Tuple) = signed > 0 ? tuple : (last(tuple), first(tuple))
+
+function minimum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    # The minimum of d.ρ[i] will minimize d[j] if d.σ[i,j] is negative. 
+    # Otherwise the maximum will minimize d[j]
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    return mins
+end
```

Comment:
This seems inefficient and incorrect - it seems it does not take into account the shift?

Reply:
I'm all ears if you have a more efficient way to do it (although you're right that I forgot about the shift).